lxd/device/filters/disk.go

```diff
@@ -0,0 +1,39 @@
+package filters
+
+// IsDisk evaluates whether or not the given device is of type disk.
+func IsDisk(device map[string]string) bool {
+	return device["type"] == "disk"
+}
+
+// IsFilesystemDisk evaluates whether or not the given device contains a filesystem.
+// A filesystem disk device always holds a target path.
+// It returns true for custom volumes of type filesystem, root volumes and directory shares.
+func IsFilesystemDisk(device map[string]string) bool {
```

Comment:
The usage of this is to replace:

```go
if dev["type"] != "disk" || dev["path"] == "" {
     continue
}
```

VM root disks are an exception to this because they are not mountable, nor are they filesystem volumes, and yet they have `path=/` property in their associated disk device.

Its probably OK for now, but we'll see if we run into issues with this down the line as we switch over to it.